scripts/glTFExport.py

```diff
@@ -400,115 +402,77 @@ def _getMaterial(self):
     @timeit
     def _getMeshData(self):
         maya.cmds.select(self.maya_node)
-        selList = OpenMaya.MSelectionList()
-        OpenMaya.MGlobal.getActiveSelectionList(selList)
-        meshPath = OpenMaya.MDagPath()
-        selList.getDagPath(0, meshPath)
-        meshIt = OpenMaya.MItMeshPolygon(meshPath)
-        meshFn = OpenMaya.MFnMesh(meshPath)
-        do_color = False
-        if meshFn.numColorSets():
-            do_color=True
-            print "doing color"
-        indices = []
-        positions = [None]*meshFn.numVertices()
-        normals = [None]*meshFn.numVertices()
-        all_colors = [None]*meshFn.numVertices()
-        uvs = [None]*meshFn.numVertices()
-        ids = OpenMaya.MIntArray()
-        tracker = {}
-        points = OpenMaya.MPointArray()
+        sel_list = om.MGlobal.getActiveSelectionList()
+        mesh_path = sel_list.getDagPath(0)
+        mesh_fn = om.MFnMesh(mesh_path)
+        do_color = bool(mesh_fn.numColors())
         if do_color:
-            vertexColorList = OpenMaya.MColorArray()
-            meshFn.getFaceVertexColors(vertexColorList)
-        #meshIt.hasUVs()
-        normal = OpenMaya.MVector()
-        face_verts = OpenMaya.MIntArray()
-        polyNormals = OpenMaya.MFloatVectorArray()
-        meshFn.getNormals(polyNormals)
-        uv_util = OpenMaya.MScriptUtil()
-        uv_util.createFromList([0,0], 2 )
-        uv_ptr = uv_util.asFloat2Ptr()
+            vertex_colors = mesh_fn.getFaceVertexColors()
+
         bbox = BoundingBox()
-        while not meshIt.isDone():
-            meshIt.getTriangles(points, ids)
-            #meshIt.getUVs(u_list, v_list)        
-            meshIt.getVertices(face_verts)
-            for x in range(0, ids.length()):
-                indices.append(ids[x])
-                pos = (points[x].x, points[x].y, points[x].z)
-                
-                local_vert_id = getFaceVertexIndex(face_verts, ids[x])
-                #print "local vert:", local_vert_id
-                norm_id = meshIt.normalIndex(local_vert_id)
-                #meshIt.getNormal(local_vert_id, normal)
-                normal = polyNormals[norm_id]
-                norm = (normal.x, normal.y, normal.z)
-                #print norm
-                meshIt.getUV(local_vert_id, uv_ptr, meshFn.currentUVSetName())
-                u = uv_util.getFloat2ArrayItem( uv_ptr, 0, 0 )
-                v = uv_util.getFloat2ArrayItem( uv_ptr, 0, 1 )
-                # flip V for openGL
-                # This fails if the the UV is exactly on the border (e.g. (0.5,1))
-                # but we really don't know what udim it's in for that case.
+
+        def _iter_mesh_faces(mesh_path):
```

Comment:
Does this do something more than mesh_iter.next()?

Reply:
Kind of, it doesn't rely on checking `isDone` every step through the loop, and is guaranteed to stop after one loop through the mesh. Mostly its just a style difference.